### PR TITLE
Rewrite HandleMessage code for components

### PR DESCRIPTION
**Pull request by InquisitivePenguin**
Corresponding PR for engine: https://github.com/space-wizards/RobustToolbox/pull/1028

### What this PR does
Reworks `HandleMessage` within component code, splitting it into two separate functions `HandleMessage` and `HandleNetworkMessage`. Also removes unused dependencies within components that no longer need to fetch the player session (since it is provided with `HandleNetworkMessage`).

### CODE REVIEW

File: Content.Client/GameObjects/Components/Construction/ConstructorComponent.cs [hunk 1]
```diff
@@ -33,9 +34,9 @@ public override void Initialize()
             Owner.GetComponent<ITransformComponent>();
         }
 
-        public override void HandleMessage(ComponentMessage message, INetChannel netChannel = null, IComponent component = null)
+        public override void HandleNetworkMessage(ComponentMessage message, INetChannel channel, ICommonSession session = null)
```
Comment (PJB3005) [inline Content.Client/GameObjects/Components/Construction/ConstructorComponent.cs:37]: `PlayerAttachedMsg` and `PlayerDetachedMsg` are raised locally so you'll need to separate these out.

File: Content.Client/GameObjects/Components/Items/ClientHandsComponent.cs [hunk 2]
```diff
@@ -144,10 +145,35 @@ public override void ExposeData(ObjectSerializer serializer)
...
+                    _gui.UpdateHandIcons();
+                    break;
+
+                case PlayerDetachedMsg _:
+                    _gui.Parent?.RemoveChild(_gui);
+                    break;
+            }
+        }
+
+        public override void HandleNetworkMessage(ComponentMessage message, INetChannel channel, ICommonSession session = null)
```
Comment (PJB3005) [inline Content.Client/GameObjects/Components/Items/ClientHandsComponent.cs:174]: this only gets raised locally so you don't need to duplicate it.

File: Content.Client/GameObjects/Components/Mobs/CameraRecoilComponent.cs [hunk 3]
```diff
@@ -62,9 +63,21 @@ public override void Kick(Vector2 recoil)
             _updateEye();
         }
 
-        public override void HandleMessage(ComponentMessage message, INetChannel netChannel = null, IComponent component = null)
+        public override void HandleMessage(ComponentMessage message, IComponent component)
```
Comment (PJB3005) [inline Content.Client/GameObjects/Components/Mobs/CameraRecoilComponent.cs:66]: This only gets raised on the server so you can remove this one.

File: Content.Client/GameObjects/Components/Mobs/ClientOverlayEffectsComponent.cs [hunk 4]
```diff
@@ -53,7 +54,20 @@ public override void OnAdd()
...
+                case PlayerAttachedMsg _:
+                    SetOverlay(_currentEffect);
+                    break;
+                case PlayerDetachedMsg _:
+                    RemoveOverlay();
+                    break;
+            }
+        }
+
+        public override void HandleNetworkMessage(ComponentMessage message, INetChannel channel, ICommonSession session = null)
```
Comment (PJB3005) [inline Content.Client/GameObjects/Components/Mobs/ClientOverlayEffectsComponent.cs:70]: Local only, remove dupe.

**Comment by InquisitivePenguin** [comment]
Sorry about the redundancy, I forgot to go back to some of the files I was working on earlier where I was overloading both methods. :P

**Pull request merged**